packages/wrap/tests/wrapSchema.test.ts

```diff
@@ -0,0 +1,84 @@
+import { execute, parse } from 'graphql';
+import { makeExecutableSchema } from '@graphql-tools/schema';
+import { wrapSchema } from '../src/wrapSchema';
+
+const typeDefs = /* GraphQL */`
+  type Query {
+    boo: String!
+  }
+`
+
+const resolvers = {
+  Query: {
+    boo() {
+      return 'boo';
+    },
+  },
+};
+
+test('should run executor once on a single root field', async () => {
+  const query = parse(/* GraphQL */`
+    {
+      boo
+    }
+  `);
+
+  const executor = jest.fn();
+  const innerSchema = makeExecutableSchema({
+    resolvers,
+    typeDefs,
+  });
+  const schema = wrapSchema({
+    schema: innerSchema,
+    executor({ document, context, variables }) {
+      executor();
+      return execute({
+        schema: innerSchema,
+        document,
+        contextValue: context,
+        variableValues: variables
+      }) as any;
+    }
+  })
+
+  await execute({
+    schema,
+    document: query,
+  });
+
+  expect(executor).toBeCalledTimes(1);
+});
+
+test('should run executor once on multiple root fields', async () => {
+  const query = parse(/* GraphQL */`
+  {
+    foo: boo
+    bar: boo
+  }
+`);
+
+  const executor = jest.fn();
+  const innerSchema = makeExecutableSchema({
+    resolvers,
+    typeDefs,
+  });
+  const schema = wrapSchema({
+    schema: innerSchema,
+    executor({ document, context, variables }) {
+      executor();
+      return execute({
+        schema: innerSchema,
+        document,
+        contextValue: context,
+        variableValues: variables
+      }) as any;
+    }
+  })
+
+  await execute({
+    schema,
+    document: query,
+  });
+
+  expect(executor).toBeCalledTimes(1);
```

Comment:
This is what we would like to have, but it turns out it is *2* times at the moment

Reply:
@maapteh I created a shared session for graphql-modules to fix it https://github.com/Urigo/graphql-modules/pull/1298/files#diff-99b394cbed0322f9673efc2cc4e1b5f5R347-R370